### PR TITLE
Add installation for slider itself.

### PR DESCRIPTION
**Pull request by ebd2**
This adds the capability to install and uninstall slider from a node on the
cluster and adds interactive configuration for the minimum config options
needed to install presto and run slider for further invocations.

Product tests are a WIP, I'll add them in another commit on the branch.

Besides the parts directly pertaining to installing slider, I've factored some of the validators out of topology.py into a module of their own under util. This is reflected in a refactoring of the tests for the validators as well. Having modified those files, I also took care of some PEP-8 errors that came up when I ran `make lint`.

I'm not sure I love how the _SLIDER_CONFIG has ended up. I've used this pattern in the past to separate the "interesting" parts of user interactions from the structure of the control flow, and had it come out cleaner. If somebody has a better idea, I'm open to it.

I would appreciate review from @anusudarsan, @rschlussel, and @cawallin. In the meantime, I'm going to start a broader refactoring of the test code to extract the standalone-specific parts out of BaseProductTest.

### CODE REVIEW

File: prestoadmin/slider/config.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+from fabric.operations import prompt
+
+from prestoadmin import config
+from prestoadmin.config import ConfigFileNotFoundError
+
+from prestoadmin.util.validators import validate_host, validate_port, \
+    validate_username, validate_can_connect, validate_can_sudo
+
+SLIDER_CONFIG_LOADED = 'slider_config_loaded'
+SLIDER_CONFIG_PATH = '/etc/opt/prestoadmin/slider/config.json'
```
Comment (anusudarsan) [inline prestoadmin/slider/config.py]: Is there an existing const we can re-use for '/etc/opt/prestoadmin' ?

File: prestoadmin/slider/config.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+            validate_args = [conf[k] for k in self.validate_keys]
+            if self.validate(*validate_args):
+                break
+            print (self.validate_failed_text % self.validate_keys) % conf
+
+
+_SLIDER_CONFIG = [
+    MultiConfigItem([
+        SingleConfigItem(HOST, 'Enter the hostname for the slider master:',
```
Comment (anusudarsan) [inline prestoadmin/slider/config.py]: maybe default it to localhost?

File: prestoadmin/slider/server.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from fabric.api import env, task, abort
+from fabric.decorators import runs_once
+from fabric.operations import put, sudo
+from fabric.tasks import execute
+
+from prestoadmin.slider.config import requires_conf, DIR, SLIDER_MASTER
+
+from prestoadmin.util.fabricapi import get_host_list, task_by_rolename
+
+__all__ = ['slider_install', 'slider_uninstall']
```
Comment (anusudarsan) [inline prestoadmin/slider/server.py]: the tasks names should be just install and uninstall.

File: prestoadmin/slider/server.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+              (slider_tarball, slider_dir, env.host))
+
+    sudo('gunzip -c %s | tar -x -C %s --strip-components=1 && rm -f %s' %
+         (slider_file, slider_dir, slider_file))
+
+
+@task
+@requires_conf
+@task_by_rolename(SLIDER_MASTER)
+def slider_uninstall():
```
Comment (anusudarsan) [inline prestoadmin/slider/server.py]: add docstring

File: prestoadmin/slider/config.py (hunk 1 above)
Comment (rschlussel) [inline prestoadmin/slider/config.py]: Also constant should go in utils/constants.py

**Comment by rschlussel** [comment]
looks good otherwise.  +1 for the validator refactoring

File: prestoadmin/slider/server.py (hunk 3 above)
Comment (ebd2) [inline prestoadmin/slider/server.py]: I'll fix this once the mode-switching is implemented.

File: prestoadmin/slider/config.py (hunk 1 above)
Comment (ebd2) [inline prestoadmin/slider/config.py]: I've fixed my usage here, and I'll put up a separate PR to fix it globally in the tests.

File: prestoadmin/slider/config.py (hunk 2 above)
Comment (ebd2) [inline prestoadmin/slider/config.py]: I like it. Done.

File: tests/product/yarn_slider/test_slider_installation.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+from prestoadmin.slider.config import SLIDER_CONFIG_PATH, HOST, DIR, \
+    SLIDER_USER, _SLIDER_CONFIG
+
+SLIDER_DIST_FILENAME = 'slider-assembly-0.80.0-incubating-all.tar.gz'
+
+
+class TestSliderInstallation(BaseProductTestCase):
+    def setUp(self):
+        super(TestSliderInstallation, self).setUp()
+        # For the moment, we still need to topology loaded. This will change
```
Comment (rschlussel) [inline tests/product/yarn_slider/test_slider_installation.py]: fix this to be a sentence

**Comment by rschlussel** [comment]
looks good

**Comment by anusudarsan** [comment]
lgtm

File: tests/product/yarn_slider/test_slider_installation.py (hunk 5 above)
Comment (ebd2) [inline tests/product/yarn_slider/test_slider_installation.py]: Out of date comment; removed. This actually will run without the topology loaded, so I switched it to a PA_ONLY_CLUSTER already.

**Pull request closed without merging**